clang/include/clang/Analysis/FlowSensitive/DataflowAnalysisContext.h

```diff
@@ -108,6 +108,16 @@ class DataflowAnalysisContext {
   /// A null `PointeeType` can be used for the pointee of `std::nullptr_t`.
   PointerValue &getOrCreateNullPointerValue(QualType PointeeType);
 
+  /// Adds `Constraint` to current and future flow conditions in this context.
+  ///
+  /// The global condition must contain only flow-insensitive information, i.e.
```

Comment:
I wonder if `GlobalConstraint` is a bit vague. How about something like `addFlowInsensitiveInvariant`. This makes part of the comment unnecessary.

Reply:
This is a much more accurate descriptor, but IME "flow insensitive" is not good as an actual name. We used "non-flow-sensitive" extensively in nullability and gradually phasing it out in favor of the "type" metaphor, as after many months reading it still requires "deciphering". Mostly it's too long, it's a negative definition, and too technical I think.
The parallel between `addFlowConditionConstraint` and `addFlowInsensitiveInvariant` is also IMO not clear enough. (I'd be in favor of dropping "Constraint" from both, but that's a different change)

I've tried `Common` instead of `Global` which might more clearly emphasize the sharing that happens here - WDYT?
If you feel strongly about `FlowInsensitive` I can use that, it might be more consistent with the framework's style.

Reply:
perhaps just `addInvariant`. The term "invariant" kind of says it all and (at least in the literature) types are frequently referrered to as specifying program invariants.

Reply:
I don't have strong feelings about either of the options. I think I have a slight preference of Invariant over Constraint.  But I am not opposed to any of the alternatives.